### PR TITLE
Include timestamp footer in the dashboard

### LINKED ISSUE #221: Show 'generated time' in the dashboard
It is useful to show a 'generated at' timestamp in the dashboard, e.g., in a footer?

Reason: When viewing a publishing dashboard, the timestamp can confirm that you are viewing the correct version of the dashboard.

### PR DESCRIPTION
**Pull request by AdityaA1998**
Fixes #221 
```
When reviewing multiple dashboards, it may be possible that user gets
confused with the version of the dashboard.

To assist the user, ensure that he/she is viewing the correct version,
let's include a generated timestamp in the footer of the dashboard.
```

### CODE REVIEW

File: src/main/java/reposense/model/Summary.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package reposense.model;
+
+import java.util.List;
+
+/**
+ * Stores the summary.json file values.
```
Comment (yong24s) [inline src/main/java/reposense/model/Summary.java]: /**
 * Represents the structure of a summary.json in reposense-report folder.
 */

File: src/main/java/reposense/report/ReportGenerator.java [hunk 2]
```diff
@@ -31,7 +31,8 @@
      *
      * @throws IOException if templateZip.zip does not exists in jar file.
      */
-    public static void generateReposReport(List<RepoConfiguration> configs, String outputPath) throws IOException {
+    public static void generateReposReport(List<RepoConfiguration> configs, String outputPath,
+                                           String generationDate) throws IOException {
```
Comment (damithc) [inline src/main/java/reposense/report/ReportGenerator.java]: Shouldn't this be 8 space indentation?

**Comment by damithc** [comment]
Also, check to confirm the codacy warnings are not applicable.

File: frontend/src/static/js/main.js [hunk 3]
```diff
@@ -13,6 +13,7 @@ window.app = new window.Vue({
     isTabAuthorship: false,
     tabInfo: {},
     tabAuthorship: {},
+    creationDate: '',
```
Comment (eugenepeh) [inline frontend/src/static/js/main.js]: using double quote instead

File: frontend/src/static/js/api.js [hunk 4]
```diff
@@ -29,9 +29,10 @@ window.api = {
   loadSummary(callback) {
     window.REPOS = {};
 
-    loadJSON(`${REPORT_DIR}/summary.json`, (repos) => {
+    loadJSON(`${REPORT_DIR}/summary.json`, (data) => {
       const names = [];
-      repos.forEach((repo) => {
+      app.creationDate = data.dashboardGeneratedTime;
```
Comment (eugenepeh) [inline frontend/src/static/js/api.js]: window.app.creationDate

**Comment by eugenepeh** [comment]
Proposed message nit:
```
The user does not know when the report was generated
when he's looking at the dashboard.

Let's include a timestamp footer for the user to show him the
correct version of the dashboard
```
=>
```
When reviewing multiple dashboards, it may be possible that user gets
confused with the version of the dashboard.

To assist the user, ensure that he/she is viewing the correct version,
let's include a generated timestamp in the footer of the dashboard.
```

**Pull request merged**